### ISSUE TITLE
Latest update bricked Companion Pi setup[BUG]

### ISSUE THREAD

**Issue by jjbutler83** [bitfocus/companion#1317]
**Make sure you're on the latest bleeding edge build** (remove this section from the bug report)
*Before you report a bug, make sure you're not reporting something that have been fixed in the master branch by downloading a "latest" build from https://builds.bitfocus.io/companion/*

**This form is only for bugs related to the Companion software itself. If you are reporting a bug that you feel is limited to a specific Companion module, please make the bug report directly in that module's repository.** Most Repositories can be located within the Bitfocus Github account with the following naming convention: companion-module-*manufacturer*-*product*

**Describe the bug**
Ran sudo companion-update on my Companion Pi 4 box and now it's bricked and will not boot.

The update process produced this error:
[1/4] Resolving packages...
[2/4] Fetching packages...
error emberplus-connection@0.0.4: The engine "node" is incompatible with this module. Expected version ">=10.20.1". Got "8.12.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this                                 command.
Done in 29.06s.

I'm afraid that I've lost everything now as a result as I cannot get the system to boot successfully now.

**To Reproduce**
Run sudo companion-update command on Raspberry Pi 4 box.

**Expected behavior**
Update and work properly.

**Screenshots**

**Desktop (please complete the following information):**
 - OS
 - Companion Version [e.g. 2.0]

**Additional context**

**Comment by Julusian** [comment]
There are a few options here:
1) Fix it up to remain in the 2.1 releases. I think the following steps should force it into place
   - `git checkout v2.1.2`
   - `git submodule init`
   - `git submodule sync`
   - `git submodule update`
   - `./tools/yarn.sh`
1) Backup your config, and restore in a new installation of companion pi. I think it will be `/home/pi/companion`, but it could be somewhere else
1) Remain on the 2.2 branch, and update node. I don't quite know what steps will be needed here

I'm assigning this to myself to make a script for option 1 to both make it easier to downgrade, and to make the update script fail with a clear error and guidance on how to resolve it

**Comment by jjbutler83** [comment]
Thank you for getting back to me. I'll be interested to try that script when it's ready.
For the commands you listed under option 1. How do I use these without a script? 

When entering in SSH session I get error:
pi@CompanionPi:~ $ git checkout v2.1.2
fatal: not a git repository (or any of the parent directories): .git

I'll admit, I'm nowhere near a super user when it comes to linux/unix language and command use.

Thanks again!

**Comment by jjbutler83** [comment]
Alright, well I ended up finding a not-too-old full config backup and so I just downloaded the newest image and re-flashed the SD card and started over.

Took a few times for some reason, but was finally able to get a stable config up and running.
I have a few extra steps involved with the setup as I have a RPi 4 case with a screen built in that I have to install and config as well as the companion stuff.

Anyway, thanks again for your response!
